sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparisonSuite.scala

```diff
@@ -406,30 +407,59 @@ class UnwrapCastInBinaryComparisonSuite extends PlanTest with ExpressionEvalHelp
   }
 
   test("SPARK-46069: Support unwrap timestamp type to date type") {
-    val tsLit = Literal.create(ts1, TimestampType)
-    val tsNTZLit = Literal.create(ts1, TimestampNTZType)
-    val floorDate = Cast(tsLit, DateType, Some(conf.sessionLocalTimeZone))
-    val floorDateNTZ = Cast(tsNTZLit, DateType, Some(conf.sessionLocalTimeZone))
-    val dateAddOne = DateAdd(floorDate, Literal(1, IntegerType))
-    val dateAddOneNTZ = DateAdd(floorDateNTZ, Literal(1, IntegerType))
-    assertEquivalent(
-      castTimestamp(f7) > tsLit || castTimestampNTZ(f7) > tsNTZLit,
-      f7 > floorDate || f7 > floorDateNTZ)
-    assertEquivalent(
-      castTimestamp(f7) >= tsLit || castTimestampNTZ(f7) >= tsNTZLit,
-      f7 >= dateAddOne || f7 >= dateAddOneNTZ)
-    assertEquivalent(
-      castTimestamp(f7) === tsLit || castTimestampNTZ(f7) === tsNTZLit,
-      f7 === floorDate && f7 === dateAddOne || f7 === floorDateNTZ && f7 === dateAddOneNTZ)
-    assertEquivalent(
-      castTimestamp(f7) <=> tsLit || castTimestampNTZ(f7) <=> tsNTZLit,
-      FalseLiteral || FalseLiteral)
-    assertEquivalent(
-      castTimestamp(f7) < tsLit || castTimestampNTZ(f7) < tsNTZLit,
-      f7 < dateAddOne || f7 < dateAddOneNTZ)
-    assertEquivalent(
-      castTimestamp(f7) <= tsLit || castTimestampNTZ(f7) <= tsNTZLit,
-      f7 <= floorDate || f7 <= floorDateNTZ)
+    def doTest(tsLit: Literal, tsNTZLit: Literal, isStartOfDay: Boolean): Unit = {
+      val floorDate = Cast(tsLit, DateType, Some(conf.sessionLocalTimeZone))
+      val floorDateNTZ = Cast(tsNTZLit, DateType, Some(conf.sessionLocalTimeZone))
+      val dateAddOne = DateAdd(floorDate, Literal(1, IntegerType))
+      val dateAddOneNTZ = DateAdd(floorDateNTZ, Literal(1, IntegerType))
+      assertEquivalent(
+        castTimestamp(f7) > tsLit || castTimestampNTZ(f7) > tsNTZLit,
+        f7 > floorDate || f7 > floorDateNTZ)
+      if (isStartOfDay) {
+        assertEquivalent(
+          castTimestamp(f7) >= tsLit || castTimestampNTZ(f7) >= tsNTZLit,
```

Comment:
is there any difference between timestamp ltz and ntz? if not we can simplify the test even more
```
def doTest(ts: Literal, isStartDay: Boolean) {
  val floorDate = ...
  val dateAddOne = ...
  ....
}
```
then the caller side
```
doTest(ltzLit, ...)
doTest(ntzLit, ...)
```

Reply:
Timestamp and TimestampNTZ are different.
We can create a TimestampNTZ from `LocalDateTime.of(2023, 12, 13, 0, 0, 0, 0)` where `isStartOfDay` is true, but I did not find a way to create a Timestamp from LocalDateTime or ZonedDateTime where  the `isStartOfDay` is true.

Reply:
why must from `LocalDateTime`? The test function takes a `Literal`, and I believe there are many ways to create a ltz literal as a start of day.

Reply:
Done